docs/vector_store.ipynb

```diff
@@ -1,79 +1,356 @@
+    "\n",
+    "loader = TextLoader(\"./state_of_the_union.txt\")\n",
+    "documents = loader.load()\n",
+    "text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)\n",
+    "docs = text_splitter.split_documents(documents)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Add Documents to the Vector Store\n",
+    "\n",
+    "After text preparation and embedding generation, the following methods insert them into the Redis vector store."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Method 1: Classmethod for Direct Insertion\n",
+    "\n",
+    "This approach combines embedding creation and insertion into a single step using the from_documents classmethod:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain_community.embeddings.fake import FakeEmbeddings\n",
+    "\n",
+    "embeddings = FakeEmbeddings(size=128)\n",
+    "redis_client = redis.from_url(\"redis://127.0.0.1:6379\")\n",
+    "rvs = RedisVectorStore.from_documents(\n",
+    "    docs, embedding=embeddings, client=redis_client, index_name=\"my_vector_index\"\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Method 2: Instance-Based Insertion\n",
+    "This approach offers flexibility when working with a new or existing RedisVectorStore:\n",
+    "\n",
+    "* [Optional] Create a RedisVectorStore Instance: Instantiate a RedisVectorStore object for customization. If you already have an instance, proceed to the next step.\n",
+    "* Add Text with Metadata: Provide raw text and metadata to the instance. Embedding generation and insertion into the vector store are handled automatically."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rvs = RedisVectorStore(\n",
+    "    client=redis_client, index_name=\"my_vector_index\", embedding_service=embeddings\n",
+    ")\n",
+    "ids = rvs.add_texts(\n",
+    "    texts=[d.page_content for d in docs], metadatas=[d.metadata for d in docs]\n",
+    ")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Perfor a Similarity Search (KNN)\n",
+    "\n",
+    "With the vector store populated, it's possible to search for text semantically similar to a query.  Here's how to use KNN (K-Nearest Neighbors) with default settings:\n",
+    "\n",
+    "* Formulate the Query: A natural language question expresses the search intent (e.g., \"What did the president say about Ketanji Brown Jackson\").\n",
+    "* Retrieve Similar Results: The `similarity_search` method finds items in the vector store closest in meaning to the query."
```

Comment:
"meaning of the query"

Reply:
I still feel that "to" is more appropriate. rephrased it as "closest to the query in meaning". WDYT?